addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/ApplianceGroheOndusHandler.java

```diff
@@ -0,0 +1,247 @@
+        super.channelLinked(channelUID);
+
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+        updateChannel(channelUID, appliance, getLastMeasurement(appliance));
+    }
+
+    public void updateChannels() {
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+
+        Measurement measurement = getLastMeasurement(appliance);
+        getThing().getChannels().forEach(channel -> updateChannel(channel.getUID(), appliance, measurement));
+
+        int pollingInterval = getPollingInterval(appliance);
+        scheduler.schedule(this::updateChannels, pollingInterval, TimeUnit.SECONDS);
+    }
+
+    private int getPollingInterval(Appliance appliance) {
+        if (config.pollingInterval > 0) {
+            return config.pollingInterval;
+        }
+        return appliance.getConfig().getMeasurementTransmissionIntervall();
+    }
+
+    private void updateChannel(ChannelUID channel, Appliance appliance, Measurement measurement) {
+        String channelName = channel.getIdWithoutGroup();
+        State newState;
+        switch (channelName) {
+            case CHANNEL_NAME:
+                newState = new StringType(appliance.getName());
+                break;
+            case CHANNEL_PRESSURE:
+                newState = new QuantityType<>(measurement.getPressure(), SmartHomeUnits.BAR);
+                break;
+            case CHANNEL_TEMPERATURE:
+                newState = new QuantityType<>(measurement.getTemperatureGuard(), SIUnits.CELSIUS);
+                break;
+            case CHANNEL_VALVE_OPEN:
+                newState = getValveOpenType(appliance);
+                break;
+            default:
+                throw new IllegalArgumentException("Channel " + channel + " not supported.");
+        }
+        updateState(channel, newState);
+    }
+
+    private OnOffType getValveOpenType(Appliance appliance) {
+        OndusService service = getOndusService();
+        if (service == null) {
+            return OnOffType.ON;
+        }
+        Optional<ApplianceCommand> commandOptional;
+        try {
+            commandOptional = service.getApplianceCommand(appliance);
+        } catch (IOException e) {
+            logger.debug("Could not get appliance command", e);
+            return OnOffType.ON;
```

Comment:
If you cannot retrieve a current state, you must not send any channel update either.

Reply:
Done!